File: fundamentos/desafio-js/src/questions/lowest-balance-agency.ts
import { IAccount } from "../i-account";

/** A agência com o menor saldo é a: */
export default function lowestBalanceAgency(...accounts: Array<IAccount>) {
  const accs = new Map<number, number>();

  for (const acc of accounts) {
    accs.set(acc.agencia, (accs.get(acc.agencia) || 0) + acc.balance);
  }

  const acc = Array.from(accs.entries())
    .sort((a, b) => a[1] - b[1])[0];

  console.info(`A agência com o menor saldo é a: ${acc[0]}, $${acc[1]}`);
}

File: fundamentos/desafio-dart/lib/questions/lowest_balance_agency.dart
import 'package:desafio_dart/domains/account.dart';

/// A agência com o menor saldo é a:
void lowestBalanceAgency(List<Account> accounts) {
  final accs = Map<int, double>();

  for (var acc in accounts) {
    accs[acc.agencia] = (accs[acc.agencia] ?? 0) + acc.balance;
  }

  final sorted = accs.entries.toList();

  sorted.sort((a, b) => a.value.compareTo(b.value));

  final acc = sorted.first;

  print('A agência com o menor saldo é a: ${acc.key}, \$${acc.value}');
}
